### PR TITLE
(Hopeful) Fix of the None Gradients for Quantize.embedding when using STE and ROTATION_TRICK

### PR DESCRIPTION
**Pull request by karapostK**
In the case of STE and ROTATION_TRICK are used, the returned embedding from Quantize have emb.detach().

This has no impact on the reconstruction loss, however there is no gradient update when using the rqvae_loss, resulting in Quantize.embeddings that are not updated during training.

A dirty solution that worked for STE (and seems to work for ROTATION_TRICK and GUMBEL_SOFTMAX) is to simply return the original embeddings to compute the update from rqvae_loss.

The code can be simplified and possibly refactored, however this depends on the specific codebook gradient update according to each of the loss. Unfortunately, I can be sure that the gradient update is right only for STE but not for the other methods that I'm not familiar with.

Hopefully the changes help you with the project!

### CODE REVIEW

File: modules/rqvae.py [hunk 1]
```diff
@@ -129,17 +132,18 @@ def get_semantic_ids(
     def forward(self, batch: SeqBatch, gumbel_t: float) -> RqVaeComputedLosses:
         x = batch.x
         quantized = self.get_semantic_ids(x, gumbel_t)
-        embs, residuals = quantized.embeddings, quantized.residuals
-        x_hat = self.decode(embs.sum(axis=-1))
+        app_embds, org_embs, residuals = quantized.approx_embeddings,quantized.origin_embeddings, quantized.residuals
```
Comment (EdoardoBotta) [inline modules/rqvae.py:135]: I'm thinking returning org_embs here and handling the detach calls in this function might be cleaner

Comment (karapostK) [reply modules/rqvae.py:135]: I was looking at [Karapathy's code](https://github.com/karpathy/deep-vector-quantization/blob/main/dvq/model/quantize.py) and it seems he computes the non-reconstruction loss (diff) inside the Quantize class. Maybe a similar style would be even cleaner? Haven't looked into GUMBEL and the Rotation trick so I'm not sure what's the flow of gradient there

Comment (EdoardoBotta) [reply modules/rqvae.py:135]: I agree. Adding a loss field to the `QuantizeOutput` and computing the loss inside the `Quantize` module is probably the way to go.
